### ISSUE TITLE
Electron 28 support

### ISSUE THREAD

**Issue by RubberDuckShobe** [smartfrigde/armcord#534]
**Is your feature request related to a problem? Please describe.**
Somewhere between Electron 22 and 26, *something* broke and it caused me to be unable to use Electron apps natively under Wayland. Trying to do so causes the GPU process to crash, like this:
```
[92719:1215/205221.804328:ERROR:gbm_wrapper.cc(253)] Failed to export buffer to dma_buf: No such file or directory (2)
[92719:1215/205221.804438:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=SCANOUT
[92719:1215/205221.804611:ERROR:gbm_wrapper.cc(253)] Failed to export buffer to dma_buf: No such file or directory (2)
[92719:1215/205221.804668:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=GPU_READ
[92719:1215/205221.804714:ERROR:shared_image_factory.cc(926)] CreateSharedImage: could not create backing.
[92719:1215/205221.804764:ERROR:shared_image_factory.cc(758)] DestroySharedImage: Could not find shared image mailbox
[92719:1215/205221.804858:ERROR:gpu_service_impl.cc(1089)] Exiting GPU process because some drivers can't recover from errors. GPU process will restart shortly.
[92682:1215/205221.816605:ERROR:gpu_process_host.cc(991)] GPU process exited unexpectedly: exit_code=8704
```
It retries this a few times, but ultimately gives up.
I'm on Arch, running Hyprland with an RTX 2070 and `nvidia-dkms` drivers.

*On Electron 28 however, it works fine again! No error messages!*
Armcord doesn't work on Electron 28 though, it simply shows a grey window after loading.

**Describe the solution you'd like**
I'd like ArmCord to support Electron 28.

**Describe alternatives you've considered**
Using XWayland (performance really isn't great with that), or dropping down to an older electron version (been a while since I last did that, but iirc it breaks screenshare?)

**Additional context**
The dev tools say this:
```
VM157 renderer_init:2 Unable to load preload script: /usr/share/armcord/app.asar/ts-out/preload/preload.js
(anonymous) @ VM157 renderer_init:2
VM157 renderer_init:2 DOMException: Failed to read the 'localStorage' property from 'Window': Storage is disabled inside 'data:' URLs.
    at Object.<anonymous> (/usr/share/armcord/app.asar/ts-out/preload/preload.js:37:8)
    at Object.<anonymous> (/usr/share/armcord/app.asar/ts-out/preload/preload.js:130:3)
    at Module._compile (node:internal/modules/cjs/loader:1271:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1326:10)
    at Module.load (node:internal/modules/cjs/loader:1126:32)
    at Module._load (node:internal/modules/cjs/loader:967:12)
    at l._load (node:electron/js2c/asar_bundle:2:13642)
    at o._load (node:electron/js2c/renderer_init:2:2455)
    at node:electron/js2c/renderer_init:2:33274
    at node:electron/js2c/renderer_init:2:33726
(anonymous) @ VM157 renderer_init:2
VM157 renderer_init:2 Electron renderer.bundle.js script failed to run
(anonymous) @ VM157 renderer_init:2
VM157 renderer_init:2 Error: An object could not be cloned.
    at t.ipcRendererInternal.send (VM157 renderer_init:2:11488)
    at VM157 renderer_init:2:33364
    at VM157 renderer_init:2:33726
    at ___electron_webpack_init__ (VM157 renderer_init:2:33730)
    at VM157 renderer_init:2:33853
    at BuiltinModule.compileForInternalLoader (VM10 loaders:336:7)
    at BuiltinModule.compileForPublicLoader (VM10 loaders:272:10)
    at loadBuiltinModule (VM124 helpers:57:9)
    at Module._load (VM127 loader:943:15)
    at l._load (VM149 asar_bundle:2:13642)
(anonymous) @ VM157 renderer_init:2
```
Since its errors seem to be related to the preload script and `localStorage`, I'm going to hazard a guess and say it might be related to [this line.](https://github.com/ArmCord/ArmCord/blob/04b3d2f371a0f7b2364ea0bba682f9b0819ed444/src/preload/preload.ts#L13)

**Comment by Vendicated** [comment]
see https://github.com/Vencord/Vesktop/pull/237

**Comment by RubberDuckShobe** [comment]
oh my fucking god lmfao

**Comment by smartfrigde** [comment]
Fixed in 3.2.6
